api/src/main/java/org/pmiops/workbench/billing/FreeTierBillingService.java

```diff
@@ -295,4 +300,42 @@ public double getUserFreeTierDollarLimit(DbUser user) {
     return Optional.ofNullable(user.getFreeTierCreditsLimitDollarsOverride())
         .orElse(workbenchConfigProvider.get().billing.defaultFreeCreditsDollarLimit);
   }
+
+  /**
+   * Set a Free Tier dollar limit override value for this user, but only if the value to set differs
+   * from the system default or the user has an existing override. If the user has no override and
+   * the value to set it equal to the system default, retain the system default so this user's quota
+   * continues to track it.
+   *
+   * <p>If this is greater than the user's total cost, set their workspaces to active. Note:
+   * lowering the limit below total cost will NOT set the workspaces to inactive.
+   * checkFreeTierBillingUsage() will do this as part of the next cron run.
+   *
+   * @param user the user as represented in our database
+   * @param newDollarLimit the US dollar amount, represented as a double
+   * @return whether an override was set
+   */
+  public boolean maybeSetDollarLimitOverride(DbUser user, double newDollarLimit) {
+    final Double previousLimitMaybe = user.getFreeTierCreditsLimitDollarsOverride();
+
+    if (previousLimitMaybe != null
+        || costsDiffer(
+            newDollarLimit, workbenchConfigProvider.get().billing.defaultFreeCreditsDollarLimit)) {
+
+      // TODO: prevent setting this limit directly except in this method?
```

Comment:
Best way to do that is with a separate little service that only exposes one entry point.

Reply:
That's effectively what we're doing already, at the Service or Controller level.  I'm imagining something stricter which we probably can't do, like "only FreeTierBillingService can call this DbUser accessor"